verify-service/src/main/java/io/mosip/verifyservice/services/VerifiablePresentationSubmissionServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+import io.mosip.verifycore.enums.SubmissionStatus;
+import io.mosip.verifycore.enums.VerificationStatus;
+import io.mosip.verifycore.exception.VerificationFailedException;
+import io.mosip.verifycore.models.VcResult;
+import io.mosip.verifycore.models.VpSubmission;
+import io.mosip.verifycore.shared.Constants;
+import io.mosip.verifycore.spi.VerifiablePresentationSubmissionService;
+import io.mosip.verifyservice.repository.AuthorizationRequestCreateResponseRepository;
+import io.mosip.verifyservice.repository.VpSubmissionRepository;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.mosip.verifycore.utils.SecurityUtils.*;
+import static io.mosip.verifycore.utils.Utils.getVerificationStatus;
+
+@Service
+public class VerifiablePresentationSubmissionServiceImpl implements VerifiablePresentationSubmissionService {
+
+    @Autowired
+    AuthorizationRequestCreateResponseRepository authorizationRequestCreateResponseRepository;
+    @Autowired
+    VpSubmissionRepository vpSubmissionRepository;
+
+    @Override
+    public VpSubmissionResponseDto submit(VpSubmissionDto vpSubmissionDto) {
+        new Thread(() -> {
+            processSubmission(vpSubmissionDto);
+        }).start();
+        return new VpSubmissionResponseDto("", "", "");
+
+    }
+
+    private void processSubmission(VpSubmissionDto vpSubmissionDto) {
+        JSONObject vpProof = new JSONObject(vpSubmissionDto.getVpToken()).getJSONObject(Constants.KEY_PROOF);
+        String jws = getFormattedJws(vpProof.getString(Constants.KEY_JWS));
+        String publicKeyPem = vpProof.getString(Constants.KEY_VERIFICATION_METHOD);
+
+        //TODO: Dynamic algo type
+        try {
+            Algorithm algorithm = Algorithm.RSA256(getPublicKeyFromString(publicKeyPem), null);
+            JWTVerifier verifier = JWT.require(algorithm).build();
+            verifier.verify(jws);
+
+            JSONArray verifiableCredentials = new JSONObject(vpSubmissionDto.getVpToken()).getJSONArray(Constants.KEY_VERIFIABLE_CREDENTIAL);
+            List<VcResult> verificationResults = getVcVerificationResults(verifiableCredentials);
+            boolean combinedVerificationStatus = true;
+            for (VcResult verificationResult : verificationResults) {
+                combinedVerificationStatus = combinedVerificationStatus && (verificationResult.getVerificationStatus() == VerificationStatus.SUCCESS);
+            }
+            if (!combinedVerificationStatus) {
+                throw new VerificationFailedException();
+            }
+            vpSubmissionRepository.save(new VpSubmission(vpSubmissionDto.getState(), vpSubmissionDto.getVpToken(), vpSubmissionDto.getPresentationSubmission(), SubmissionStatus.SUCCESS));
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Please use loggers, printStackTrace should not be used.

Reply:
Done using Slf4j

Reply:
we should have mininum set of logger statements for understanding the flow.

Reply:
will be taken care as part of https://mosip.atlassian.net/browse/INJIVER-846